pkg/internal/sqlsmith/scalar.go

```diff
@@ -302,10 +303,10 @@ var compareOps = [...]tree.ComparisonOperator{
 }
 
 func makeCompareOp(s *Smither, typ *types.T, refs colRefs) (tree.TypedExpr, bool) {
-	typ, ok := s.pickAnyType(typ)
-	if !ok {
+	if f := typ.Family(); f != types.BoolFamily && f != types.AnyFamily {
 		return nil, false
 	}
+	typ = s.randScalarComparableType()
```

Comment:
So it turns out that the only reason we don't support json comparison is because we didn't add it. It works in postgres, and I was able to add it to cockroach without any trouble (we just didn't plumb some operators down, but the functionality already exists). So you can remove this whole function and array here. I'll add a PR to enable json compare, and this will just turn on. It's ok that it's producing errors for now.